### PR TITLE
Structured Content

### PR DESCRIPTION
**Pull request by reknih**
Structured Content, Tagged PDF, Name Trees, ... You name it and this PR has got it!

### CODE REVIEW

File: src/attributes.rs [hunk 1]
```diff
@@ -0,0 +1,864 @@
...
+        self.dict.pair(Name(b"Summary"), summary);
+        self
+    }
+}
+
+deref!('a, TableAttributes<'a> => Dict<'a>, dict);
+
+/// The scope of a table header cell.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum TableHeadingScope {
```
Comment (laurmaedje) [inline src/attributes.rs:798]:
```suggestion
pub enum TableHeaderScope {
```

File: src/attributes.rs [hunk 2]
```diff
@@ -0,0 +1,864 @@
...
+pub struct Attributes<'a> {
+    obj: Obj<'a>,
+}
+
+writer!(Attributes: |obj| Self { obj });
+
+impl<'a> Attributes<'a> {
+    /// Set the `/O` attribute to user-defined and start writing the `/P` array
+    /// with user properties. PDF 1.6+
+    pub fn user_properties(self) -> PropertyAttributes<'a> {
```
Comment (laurmaedje) [inline src/attributes.rs:15]: Let's talk about this personally

File: src/structure.rs [hunk 3]
```diff
@@ -190,6 +258,492 @@ impl<'a> ViewerPreferences<'a> {
...
+
+writer!(ObjectRef: |obj| {
+    let mut dict = obj.dict();
+    dict.pair(Name(b"Type"), Name(b"OBJR"));
+    Self { dict }
+});
+
+impl<'a> ObjectRef<'a> {
+    /// Write the `/Pg` attribute to specify the page some or all of this
+    /// structure element is located on.
```
[the remark this answers is not in the document]
Comment (reknih) [reply src/structure.rs:522]: `Write the `/Pg` attribute to specify the page the referenced marked content sequence is located on.`

It has to be a single page, that is true

File: src/attributes.rs [hunk 4]
```diff
@@ -0,0 +1,864 @@
...
+    /// for all extra columns. PDF 1.6+.
+    pub fn column_gap(&mut self) -> TypedArray<'_, f32> {
+        self.dict.insert(Name(b"ColumnGap")).array().typed()
+    }
+}
+
+/// Vertical Text.
+impl LayoutAttributes<'_> {
+    /// Write the `/GlyphOrientationVertical` attribute as an angle between 0 and 360. PDF 1.5+.
+    pub fn glyph_orientation_vertical(&mut self, angle: f32) -> &mut Self {
```
[the remark this answers is not in the document]
Comment (reknih) [reply src/attributes.rs:528]: Auto is the default and need thus not be handled

**Pull request merged**